plugin/build.gradle.kts

```diff
@@ -83,14 +85,26 @@ sourceSets {
     }
 }
 
+kotlin {
+    @OptIn(ExperimentalBuildToolsApi::class, ExperimentalKotlinGradlePluginApi::class)
+    compilerVersion = libs.versions.kotlin.`for`.gradle.plugin.get()
+}
+
 tasks.compileKotlin {
     compilerOptions {
         optIn.addAll(
                 "kotlinx.benchmark.gradle.internal.KotlinxBenchmarkPluginInternalApi",
                 "kotlin.RequiresOptIn",
         )
-        @Suppress("DEPRECATION")
-        apiVersion = KotlinVersion.KOTLIN_1_4 // the version of Kotlin embedded in Gradle
+        /**
+         * Those versions are configured according to https://docs.gradle.org/current/userguide/compatibility.html
+         * and the Kotlin compiler compatibility policy stating that Kotlin 1.4 is compatible with 1.5 binaries
+         */
+        @Suppress("DEPRECATION", "DEPRECATION_ERROR")
+        run {
+            languageVersion = KotlinVersion.KOTLIN_1_5
+            apiVersion = KotlinVersion.KOTLIN_1_4
+        }
```

Comment:
Could you please elaborate on this part a little more?

If I understand correctly, Kotlin compiler 2.1 will drop support for `language/apiVersion` 1.4 and 1.5. 
Also, KGP 2.1 supports Gradle 7.6.3+, which, according to the [Gradle compatibility guide](https://docs.gradle.org/current/userguide/compatibility.html), comes with Kotlin 1.7.10 and only supports Kotlin language features available in 1.4.
Therefore, to ensure the kx-benchmark plugin continues supporting Gradle 7.6.3+ projects, we use compiler version 2.0.20 for compiling the plugin. This allows us to set `languageVersion = 1.4` (or 1.5, which is compatible with 1.4 binaries).

Please correct me if I got it wrong.

My other question is: What version should apiVersion indicate? Is it the version of the embedded Kotlin in the minimal supported Gradle? For example, for Gradle 7.6.3, should it be 1.7?

Reply:
That's right. 

> My other question is: What version should apiVersion indicate? Is it the version of the embedded Kotlin in the minimal supported Gradle? For example, for Gradle 7.6.3, should it be 1.7?

`apiVersion` affects available API from Kotlin stdlib. Thus, you're right again. It should be equal to the version of Kotlin stdlib provided by Gradle at runtime.

Reply:
Thanks for the clarification!